### DISCUSSION TITLE
Reference mapping on harmony-corrected LSI embedding

### DISCUSSION THREAD

**Discussion by timorueckert** [stuart-lab/signac#711] [Q&A] [answered]
Hi Tim, hi scATAC-seq community,

I am following your scATAC-seq data integration vignette trying to perform reference mapping. To give you some background, I have one large dataset covering different stages of a cell type ex vivo and a second, smaller dataset in which this cell type was treated with different stimuli. I would like to understand, whether and which of the stimuli induce a chromatin remodeling so that the cells become more similar to any of the stages ex vivo. Since the cells are from different human individuals, especially the ex vivo dataset had to be integrated, which was quite successfully done with harmony. As there is really a quite strong donor effect, I am afraid performing reference mapping on the uncorrected embeddings rather reflects donor-specific than global signatures.

Therefore, I tried to run the reference mapping pipeline on the harmony-corrected lsi embedding, but get an error message when running:

`transfer.anchors <- FindTransferAnchors(
  reference = exvivo.atac,
  query = invitro.atac,
  reference.reduction = "harmony",
  reduction = "lsiproject",
  dims = 2:30
)
Error in vt <- vt[features, dims]: no dimnames attribute for array`

From my understanding, this is because there are no loadings for the harmony reduction, as RunHarmony was called with ProjectDim = FALSE. I do not seem to get around this by setting this to TRUE, as there is no "scaled" assay which could be passed to ProjectDim.

Do you have any suggestion as to how to make this work? Or potentially an alternative approach to tackle the question I am facing?

Thanks a lot and congratulations for generating such an invaluable resource,
Timo

**Comment by timoast** [comment, marked as answer]
Harmony corrects the low-dimensional cell embeddings, but does not compute a set of "corrected" loadings that would enable the original data to be projected onto the corrected space. Similarly for the `IntegrateEmbeddings()` function in Seurat, we currently do not have a way of computing a set of new loadings that would project the original data onto the corrected embedding space. 

I'd suggest integrating the scATAC-seq datasets (using Harmony or Seurat) to obtain a set of consistent cell-type annotations, and then running a _new_ LSI on the merged, uncorrected dataset. This will contain batch effects, but should be a more comprehensive dataset for mapping query datasets. Your hope would be that queries are highly similar to at least one of the batches in the reference, and since the reference cells are consistently annotated you should be able to accurately predict cell labels in the query.

**Reply by timorueckert** [reply]
As you suggested, I clustered and annotated the merged reference based on the harmony-corrected embedding before performing the prediction on the uncorrected LSI. The cell type prediction worked actually really well and fit to what I was hoping to see. The results were also very consistent between two different donors, so I am quite confident the predictions are rather accurate and reflect more than batch effects.

Thanks for your help! :)
